autometa/taxonomy/ncbi.py

```diff
@@ -364,27 +332,66 @@ def parse_merged(self):
             logger.debug('merged loaded')
         return merged
 
-    def is_common_ancestor(self, parent_taxid, child_taxid):
-        """Determines whether the provided taxids have a non-root common ancestor
+    def is_common_ancestor(self, taxid1, taxid2):
+        """
+        Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
-        parent_taxid : int
-            Description of parameter `parent_taxid`.
-        child_taxid : int
-            Description of parameter `child_taxid`.
+        taxid1 : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+        taxid2 : int
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         boolean
-            True if taxids share a common ancestor and False otherwise
+            True if taxids share a common ancestor else False
         """
-        ancestor_taxid = child_taxid
-        while ancestor_taxid != 1:
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
-        return False
+        taxid1_lineage = self.lineage(taxid1)
+        taxid2_lineage = self.lineage(taxid2)
+        taxid1_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid1_lineage])
+        taxid2_lineage_taxids = set([taxids.get('taxid')
+                                     for taxids in taxid2_lineage])
```

Comment:
Is it necessary to have multiple variables here? I think you could consolidate these lines and reduce unnecessary variables.

```suggestion
        taxid1_lineage_taxids = {taxids.get('taxid') 
                                 for taxids in self.lineage(taxid1)}
        taxid2_lineage_taxids = {taxids.get('taxid') 
                                 for taxids in self.lineage(taxid2)}
```